bluesky_kafka/tests/conftest.py

```diff
@@ -23,3 +39,109 @@ def bootstrap_servers(request):
 @pytest.fixture(scope="function")
 def hw(request):
     return ophyd.sim.hw()
+
+
+@pytest.fixture(scope="function")
+def msgpack_serializer(request):
+    return partial(msgpack.dumps, default=mpn.encode)
+
+
+@pytest.fixture(scope="function")
+def msgpack_deserializer(request):
+    return partial(msgpack.loads, object_hook=mpn.decode)
+
+
+@pytest.fixture(scope="function")
+def publisher(request, bootstrap_servers, msgpack_serializer):
```

Comment:
This fixture and the next one end up being very specific. I have never written a [factory as a fixture](https://docs.pytest.org/en/stable/fixture.html#factories-as-fixtures) but I think this is the right place for one. For example:
```python

@pytest.fixture
def build_test_publisher(request, bootstrap_servers, msgpack_serializer):
    def _build_test_publisher(
        topic,
        key="kafka-unit-test-key",
        producer_config={
            "acks": 1,
            "enable.idempotence": False,
            "request.timeout.ms": 5000,
        },
        serializer= msgpack_serializer,
        flush_on_stop_doc=True
    ):
        return Publisher(
            topic=topic,
            bootstrap_servers=bootstrap_servers,
            key=key,
            producer_config=producer_config,
            serializer=serializer,
            flush_on_stop_doc=flush_on_stop_doc,
        )
```

Then a test function would look something like:
```python
def test_something(build_test_publisher):
    topic_1_publisher = build_test_publisher(topic="topic_1")
    topic_2_publisher = build_test_publisher(topic="topic_2")
    ...
```

Reply:
Can we make this into an issue and add it to a later PR? Because I only need 2 publishers, maybe its okay to not have a factory?